webdriver/tests/take_element_screenshot/user_prompts.py

```diff
@@ -15,56 +15,94 @@ def take_element_screenshot(session, element_id):
         )
     )
 
+@pytest.fixture
+def check_user_prompt_closed_without_exception(session, create_dialog):
+    def check_user_prompt_closed_without_exception(dialog_type):
+        session.url = inline("<input/>")
+        element = session.find.css("input", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = take_element_screenshot(session, element.id)
+        value = assert_success(response)
+
+        assert_png(value)
+
+    return check_user_prompt_closed_without_exception
+
 
 @pytest.fixture
-def check_user_prompt_not_closed_without_exception(session, create_dialog):
-    def check_user_prompt_not_closed_without_exception(dialog_type):
+def check_user_prompt_closed_with_exception(session, create_dialog):
+    def check_user_prompt_closed_with_exception(dialog_type):
         session.url = inline("<input/>")
         element = session.find.css("input", all=False)
 
         create_dialog(dialog_type, text=dialog_type)
 
+        response = take_element_screenshot(session, element.id)
+        assert_error(response, "unexpected alert open")
+
+        # Since dismiss and notify or accept and notify would have dismissed,
+        # retrying the screenshot will succeed
         response = take_element_screenshot(session, element.id)
         value = assert_success(response)
```

Comment:
No user prompt tests for all the other commands is doing that. To stay in sync please remove this extra check. Also what you basically want is the call to `assert_dialog_handled()` here too, which plainly is doing such a check that the dialog is gone.

Reply:
I've reworked most of this change to follow closely what other user_prompts.py files are testing for. Please let me know if you see any other things that may need changing.